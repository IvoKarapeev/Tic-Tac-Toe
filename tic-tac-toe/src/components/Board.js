import Box from './Box';
import './Board.css';

const Board = ({board,onClick}) => {

    return(
        <div className='board' data-testid='board'>
            {board.map((value, idx) => {
                return <Box key={idx} value={value} onClick={() => value === null && onClick(idx)}/>
            })}
            
            </div>
        
    )
};

export default Board;