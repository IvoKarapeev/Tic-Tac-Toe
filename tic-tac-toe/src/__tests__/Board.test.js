import React from "react";
import '@testing-library/jest-dom';
import { render,screen,cleanup } from '@testing-library/react';

import Board from "../components/Board";

afterEach(() => {
    cleanup();
});

describe('Board Component Testing',() => {
    test('shoud render Board component',() => {
        const board = Array(9).fill(null)
        render(<Board board={board} onClick={() => {}}/>);
        const boardElement = screen.getByTestId('board');
        expect(boardElement).toBeInTheDocument();
    });
    test('rendered Board component with right childs',() => {
        
        let board = Array(9).fill(null);

        const onClick = () => {
        };
        
        render(<Board board={board} onClick={onClick}/>);
        const boardElement = screen.getByTestId('board');
        expect(boardElement).toMatchSnapshot();
    });
    test('rendered Board component with different props',() => {
        
        let board = ['X',  null, null, 'X',  null, null, 'X', null, null]

        const onClick = () => {
        };
        
        render(<Board board={board} onClick={onClick}/>);
        const boardElement = screen.getByTestId('board');
        expect(boardElement).toMatchSnapshot();
    });

});