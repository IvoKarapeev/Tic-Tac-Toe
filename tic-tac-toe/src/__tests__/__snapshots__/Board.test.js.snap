// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Component Testing rendered Board component with different props 1`] = `
<div
  class="board"
  data-testid="board"
>
  <button
    class="box x"
  >
    X
  </button>
  <button
    class="box o"
  />
  <button
    class="box o"
  />
  <button
    class="box x"
  >
    X
  </button>
  <button
    class="box o"
  />
  <button
    class="box o"
  />
  <button
    class="box x"
  >
    X
  </button>
  <button
    class="box o"
  />
  <button
    class="box o"
  />
</div>
`;

exports[`Board Component Testing rendered Board component with right childs 1`] = `
<div
  class="board"
  data-testid="board"
>
  <button
    class="box o"
  />
  <button
    class="box o"
  />
  <button
    class="box o"
  />
  <button
    class="box o"
  />
  <button
    class="box o"
  />
  <button
    class="box o"
  />
  <button
    class="box o"
  />
  <button
    class="box o"
  />
  <button
    class="box o"
  />
</div>
`;
